src/app/reducers/smartBanner.js

```diff
@@ -29,6 +31,20 @@ export default function(state=DEFAULT, action={}) {
       });
     }
 
+    case loginActions.LOGGED_IN: {
+      if (state.haveShownXPromo) {
+        // NOTE: we should really only be doing this
+        // in the case that we are running a "login required"
+        // variant. Since we don't have access to the full state
+        // here, that's difficult to find out.
```

Comment:
Could we dispatch an action when the login variant is shown, which we could handle in this reducer, so that we know exactly which situation we're in?